sqitchers/sqitch
# Update and restore engine versions and tests

Restore the coverage tests and update the versions tested for all the database engines.

*   Test Exasol 7.0, 7.1, and 8 with its v24 clients and update the error-checking regular expression to support it.
*   Update the Exasol test to detect apparent connection failures and try again for up to 30 seconds before failing. This lets the Exasol tests run consistently. Technique borrowed from the Vertica tests (fcd7fe1c9).
*   Use `SSLCERTIFICATE` instead of `SSLCertificate` to be consistent with the the Exasol ODBC docs.
*   Upgrade Instant Client to v23.6.24.10 and install libaio1t64 for Oracle and create a symnlink so that Instant Client can find it.
*   Upgrade SnowSQL to 1.2.21 and Snowflake ODBC to 3.5.0.
*   Upgrade the Vertica client to v24.2.0 and test on Vertica vefsions 7.2 - 24.1.
*   Switch to a new Docker image for Firebird and use it to test versions 3-5. Continue using the old one to test v2.5.
*   Update the coverage tests to clear more disk space, improving the chances that the tests will finish.
*   Test MySQL v5.5-9.1 and MariaDB v10.0-11.6
*   Add Perl 5.40 testing
*   Add Postgres 17 testing
*   Add SQLite testing for v3.43-3.47
*   Add Vertica v23-24 testing
*   Add Yugabyte v2.20, 2.21, 2.22, 2.23, 2024.1, and 2024.2 testing
*   Teach the IPC methods in Sqitch.pm to raise App::Sqitch::X exceptions instead of strings for more consistent error handling (esp. in tests).
*   Add `DBIEngine->_error_handler` and always pass it to `DBI->connect`, replacing redundant and inconsistenlty-applied `HandleError` functions.
*   Change `DBIEngineTest->run` to always expect error objects, now that the IPC methods raise them and tests use `DBIEngine->_error_handler` to get connection failure exception objects.
*   Add some hijinks to `t/lib/TestConfig.pm` to suppress warnings from Locale::Messages on newer Perls (workaround for gflohr/libintl-perl#14).
*   Fix a bunch of shellcheck warnings.
